src/client/core/utils/dom.js

```diff
@@ -364,28 +367,81 @@ export function getElementDescription (el) {
     return res.join('');
 }
 
-export function getNextFocusableElement (element, reverse) {
-    const offset       = reverse ? -1 : 1;
-    let allFocusable = sortElementsByFocusingIndex(getFocusableElements(findDocument(element)));
+export function focusNextElement (element, reverse, skipRadioGroups) {
+    return new Promise(resolve => {
+        const nextElement = getNextFocusableElement(element, reverse, skipRadioGroups);
+
+        if (!nextElement)
+            resolve();
 
-    //NOTE: in all browsers except Mozilla and Opera focus sets on one radio set from group only.
-    // in Mozilla and Opera focus sets on any radio set.
-    if (isInputElement(element) && element.type === 'radio' && element.name !== '' && !browserUtils.isFirefox) {
-        allFocusable = arrayUtils.filter(allFocusable, item => {
-            return !item.name || item === element || item.name !== element.name;
+        focusBlurSandbox.focus(nextElement, () => {
```

Comment:
If we do not want to call `focus` if `!nextElement === true` we should call return.
maybe we need test for this case

Reply:
we can simplify `focusBlurSandbox.focus(nextElement, () => resolve(nextElement));`

Reply:
`getNextFocusableElement` is not supposed to return null for now, so this extra check is required to prevent errors in future